src/localnet.ts

```diff
@@ -0,0 +1,126 @@
+import { execSync, ExecSyncOptions } from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+
+const REPO_URL: string = "https://github.com/matter-labs/local-setup.git";
+const REPO_BRANCH: string = "main"
+
+// ---------------------------------------------------------------------------------------
+// Utilities
+// ---------------------------------------------------------------------------------------
+
+function runCommand(command: string, options?: ExecSyncOptions): string {
+    const defaultOptions: ExecSyncOptions = { cwd: repoDirectory(), encoding: 'utf-8' };
+    const unifiedOptions: ExecSyncOptions = {...defaultOptions, ...options};
+    return execSync(command, unifiedOptions).toString();
+}
+
+function repoDirectory(): string {
+    const xdgStateHome = process.env.XDG_STATE_HOME || path.join(os.homedir(), ".local/state");
```

Comment:
comment on why you use XDG_STATE_HOME

Reply:
Sorry for the late response. I missed the notifications.

I decided to use `XDG_STATE_HOME` because I considered the presence or absence of this repo to be part of the CLI tool state.

I wanted to follow the  [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) in order to follow some standard that keeps the end user's `$HOME` tidy.

For reference, here is the relevant part of the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
>$XDG_STATE_HOME defines the base directory relative to which user-specific state files should be stored. If $XDG_STATE_HOME is either not set or empty, a default equal to $HOME/.local/state should be used.

The other alternative that I've considered was `XDG_DATA_HOME`
> XDG_DATA_HOME defines the base directory relative to which user-specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

I think that the other XDG base directories, `XDG_CONFIG_HOME`, `XDG_CACHE_HOME`, `XDG_RUNTIME_DIR` don't apply here.

What do you think?

Reply:
sounds good. please mention it in the code comment (so that the future readers can also understand this).

Reply:
As mentioned above, please add a comment with your explanation to the code

Reply:
Addressed @ [40b8aa8](https://github.com/matter-labs/zksync-cli/pull/16/commits/40b8aa856a316b421e689fe45752d15002704822)